agentMET4FOF/metrological_agents.py

```diff
@@ -198,3 +201,43 @@ def custom_plot_function(self, data, sender_agent, **kwargs):
         else:
             trace = go.Scatter()
         return trace
+
+class MetrologicalGeneratorAgent(MetrologicalAgent):
+    """An agent streaming a sine signal
+
+    Takes samples from an instance of :py:class:`MetrologicalSineGenerator` with sampling frequency `sfreq` and
+    signal frequency `sine_freq` and pushes them sample by sample to connected agents via its output channel.
+    """
+
+    # The datatype of the stream will be MetrologicalSineGenerator.
+    _stream: MetrologicalDataStreamMET4FOF
+
+    def init_parameters(
+        self,
+        signal: MetrologicalDataStreamMET4FOF = MetrologicalSineGenerator(),
+        **kwargs
+    ):
+        """Initialize the input data stream
+
+        Parameters
+        ----------
+        signal : MetrologicalDataStreamMET4FOF
+            the underlying signal for the generator
+        """
+        self._stream = signal
+        super().init_parameters()
+        self.set_output_data(channel="default", metadata=self._stream.metadata)
+
+    @property
+    def device_id(self):
+        return self._stream.metadata.metadata["device_id"]
+
+    def agent_loop(self):
+        """Model the agent's behaviour
+
+        On state *Running* the agent will extract sample by sample the input
+        datastream's content and push it into its output buffer.
+        """
+        if self.current_state == "Running":
+            self.set_output_data(channel="default", data=self._stream.next_sample())
+            super().agent_loop()
```

Comment:
Just add a new line at the end of the file:

```suggestion
            super().agent_loop()

```